### ISSUE TITLE
Weighted Cross Entropy Loss

### ISSUE THREAD

**Issue by antoniobalordi** [f-dangel/backpack#350]
Hi! 

Is there a way to use the Diagonal Hessian/Diag GGN for a model using a weighted Cross Entropy Loss function?

**Comment by fKunstner** [comment]
You would have to implement the derivative/Hessian of the weighted cross entropy.

The docs has an example of how to add a module https://docs.backpack.pt/en/master/use_cases/example_custom_module.html
It's slightly different because it's meant for a layer rather than a loss.

For a loss, the easiest way would be to take our code for the unweighted cross entropy loss here 
https://github.com/f-dangel/backpack/blob/master/backpack/core/derivatives/crossentropyloss.py#L16
and write `WeightedCrossEntropyDerivatives` class for your `WeightedCrossEntropy` module.

You can then plug it into backpack by doing
```
class DiagGGNWeightedCrossEntropy(DiagGGNLoss):
    def __init__(self):
        super().__init__(derivatives=WeightedCrossEntropyDerivatives())

extension = DiagGGNExact()
extension.set_module_extension(WeightedCrossEntropy, DiagGGNWeightedCrossEntropy())
```
Then the DiagGGNExact extension will work with WeightedCrossEntropy.
